server/apps/immich/src/api-v1/auth/auth.service.ts

```diff
@@ -30,11 +32,18 @@ export class AuthService {
     private oauthService: OAuthService,
     private immichJwtService: ImmichJwtService,
     @Inject(USER_REPOSITORY) userRepository: IUserRepository,
+    private configService: ImmichConfigService,
+    @Inject(INITIAL_SYSTEM_CONFIG) private config: SystemConfig,
   ) {
     this.userCore = new UserCore(userRepository);
+    this.configService.config$.subscribe((config) => (this.config = config));
   }
 
   public async login(loginCredential: LoginCredentialDto, clientIp: string): Promise<LoginResponseDto> {
+    if (!this.config.passwordLogin.enabled) {
+      throw new UnauthorizedException('Password login has been disabled');
+    }
```

Comment:
The admin will be prevented from logging in with password as well, right?

Reply:
Correct.

Reply:
I am wondering if disabling normal user login is a UX we want to implement 🤔 ?

Reply:
I'm set up with oauth personally and would love to disable password login myself. Seems probable most people with oauth enabled will turn off password login.

Reply:
If a user is autocreated via oauth, there won't be a way to log into that user with a password anyways, right?

Reply:
Correct.